src/chess/pgn.h

```diff
@@ -75,31 +77,29 @@ class PgnReader {
     bool in_comment = false;
     bool started = false;
     while (GzGetLine(file, line)) {
+      line_no_++;
```

Comment:
As far as I can see, it's possible to call `AddPgnFile` multiple times with different files. In this case, I believe `line_no_` should be reset on every call.

Also, as the class keeps the state anyway, maybe it's possible to split it into more fuctions (it kind of recembles recursive descent parsing anyway)?
And also move implementations into .cc file?